core/services/chainlink/application.go

```diff
@@ -302,6 +305,23 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 			chainSet,
 			globalLogger,
 		)
+		// TODO [relay]: add job.OffchainReporting2Relay
+		relays := relay.Relays{
+			Solana: solana.NewRelay(),
+		}
+		subservices = append(subservices, relays.Solana)
+		delegates[job.OffchainReporting2Relay] = offchainreporting2relay.NewDelegate(
+			db,
+			jobORM,
+			// TODO [relay]: solana OCR2 keys, or OCR2 keys for all relays?
```

Comment:
I think itd be OCR2 keys for all relays - we can pass an argument to ```keys ocr2 create``` like ```ed25519 || ecsda``` which would creat the appropriate keyring and then we just reference that key bundle in the spec.

Reply:
Please check it out again. I've made the Relayer impl depend on `keystore.OCR2`, which is still Ethereum specific but added a few TODOs for @RyanRHall, when he starts on Solana OCR2 keys.

Reply:
Does the relayer actually need the keyrings? If it doesn't I think the relayer shouldn't need the key bundle. I do think that the NewTransmitter method will have to accept the transmission key however

Reply:
I'm fond of it as part of the interface as then the OCR2Provider provides all required, including the OCR2 key bundle (where they come from is an impl detail). 

Yeah, the transmitter will still need the transmission key, adding next...